frontend/packages/ceph-storage-plugin/README.md

```diff
@@ -0,0 +1,31 @@
+# OCS UI Features
+
+The OCS UI requires some annotations in the OCS Operator CSV to perform various actions.
+
+Following table maps the annotation to its use case and accepted values:
+|Annotation Name|Purpose|Accepted Values |
+|---------------------------------------|---------------------------|--------|
+| `features.ocs.openshift.io/enabled`| Activate UI Features | "external", "snapshot" |
+| `external.ocs.openshift.io/validation`| Mininum required keys to be supplied by the admin to connect to an external cluster | Array of Keys that need to be validated in UI |
+||||
+
+## Enabling Features in UI
+
+UI features are activated based on the values in `features.ocs.openshift.io/enabled` annotation. The following table maps a feature and the respective annotation required to activate it.
+| Feature |Feature guard|
+|------------------------------|------------|
+| External Cluster Installation| `external` |
+| Volume Snapshots| `snapshot`
+
+#### Example
+
+       "features.ocs.openshift.io/enabled": `["external"]`
+
+## JSON validation (Independent Mode)
+
+This is used to perform validation of the JSON file uploaded by the user. The values in `external.ocs.openshift.io/validation` annotation needs to be a stringified JSON. The JSON object should have the following keys and values:
+| Key | Accepted Values | Test Performed |
+|------|-----------|-------------|
+| `secrets` | Array of String| JSON should contain all the `secrets` defined in the array. The `data.userKey` /`data.adminKey` field should be Base64 encoded and non-empty.
+|`configMaps`| Array of String| JSON should contain all the `configMaps` defined in the array. `data` field should be non-empty.
+| `storageClasses`| Array of String| JSON should contain all the `storageClasses` defined in the array. `data` field should be non-empty.
```

Comment:
I feel, we should also provide an example of the correct JSON file for reference as you have mentioned few fields like `data.userKey`, `data.adminKey`